lib/logproto/logproto-server.c

```diff
@@ -122,12 +122,20 @@ log_proto_server_init(LogProtoServer *self, LogTransport *transport, const LogPr
   self->transport = transport;
 }
 
-void
+gboolean
 log_proto_server_options_set_encoding(LogProtoServerOptions *self, const gchar *encoding)
 {
-  if (self->encoding)
-    g_free(self->encoding);
+  GIConv convert;
+
+  g_free(self->encoding);
   self->encoding = g_strdup(encoding);
+
+  /* validate encoding */
+  convert = g_iconv_open("utf8", encoding);
```

Comment:
It could be a problem... (however, I have a SUTC - small&ugly test code and it works when `tocode` and `fromcode` is the same...) Maybe different library versions...

In libiconv, they convert the character encoding names to upper case, so the solution is definitely not what @bkil-syslogng suggested.
Comment from libiconv we have:
`Before calling aliases_lookup, convert the input string to upper case`